VisualStudioPlugin/SolutionExplorerMenuCommand.cs

```diff
@@ -163,47 +156,121 @@ private void SaveToQuantConnectCallback(object sender, EventArgs e)
             var api = AuthorizationManager.GetInstance().GetApi();
             var compileStatus = api.CreateCompile(projectId);
             var compileId = compileStatus.CompileId;
             while (compileStatus.State == Api.CompileState.InQueue)
             {
-                Thread.Sleep(5000);
+                Thread.Sleep(2500);
                 compileStatus = api.ReadCompile(projectId, compileId);
             }
-            return compileStatus;
+
+            if (compileStatus.State == Api.CompileState.BuildError)
+            {
+                // Default to show Errors, now it is coming empty so use Logs. Will only show First Error || Log
+                var error = compileStatus.Errors.Count == 0 ?
+                    compileStatus.Logs.FirstOrDefault() : compileStatus.Errors.FirstOrDefault();
+                var message = "Compilation error \"" + error + "\"";
+                VsUtils.DisplayInStatusBar(_serviceProvider, message);
+                VsUtils.ShowMessageBox(_serviceProvider, "Compilation Error", error);
+                return "";
+            }
+            VsUtils.DisplayInStatusBar(_serviceProvider, "Compilation succeeded");
+            return compileStatus.CompileId;
         }
 
-        private Api.Backtest BacktestProjectOnServer(int projectId, string compileId)
+        /// <summary>
+        /// Backtests specific projectId and compileId at QuantConnect
+        /// </summary>
+        /// <param name="projectId">Target project Id</param>
+        /// <param name="compileId">Target compile Id</param>
+        /// <returns>Returns false if failed else true</returns>
+        private bool BacktestProjectOnServer(int projectId, string compileId)
         {
+            var res = true;
+            VsUtils.DisplayInStatusBar(_serviceProvider, "Backtesting project ...");
             var api = AuthorizationManager.GetInstance().GetApi();
             var backtestStatus = api.CreateBacktest(projectId, compileId, "My New Backtest");
             var backtestId = backtestStatus.BacktestId;
+
             while (!backtestStatus.Completed)
             {
                 Thread.Sleep(5000);
                 backtestStatus = api.ReadBacktest(projectId, backtestId);
+                if (backtestStatus == null)
+                {
+                    // this is happening when backtest failed and result in json is a white space char ' '
+                    break;
+                }
+            }
+            // Errors are not being transfered in response, so client can't tell if the backtest failed or not.
+            // This response error handling code will not work but should.
+            if (backtestStatus == null || backtestStatus.Errors.Count != 0)
+            {
+                VsUtils.DisplayInStatusBar(_serviceProvider, "Backtest Failed");
+                VsUtils.ShowMessageBox(_serviceProvider, "Backtest Failed", "Error when backtesting project.");
```

Comment:
`ShowErrorMessageBox` would be a nice helper

Reply:
Fixed